velero-plugins/common/restore.go

```diff
@@ -54,6 +55,29 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	} else {
 		// if the current workflow is CAM then get migration registry from backup object and set the same on annotation to use in plugins.
 		annotations[MigrationRegistry] = input.Restore.Annotations[MigrationRegistry]
+
+		// Set migmigration and migplan labels on all resources, except ServiceAccounts
+		switch input.Item.DeepCopyObject().(type) {
+		case *corev1.ServiceAccount:
+			break
+		default:
+			migMigrationLabel, exist := input.Restore.Labels[MigMigrationLabelKey]
+			if !exist {
+				p.Log.Info("migmigration label was not found on restore")
+			}
+			migPlanLabel, exist := input.Restore.Labels[MigPlanLabelKey]
+			if !exist {
+				p.Log.Info("migplan label was not found on restore")
+			}
+			labels := metadata.GetLabels()
+			if labels == nil {
+				labels = make(map[string]string)
+			}
+			labels[MigratedByLabel] = migMigrationLabel
+			labels[MigPlanLabelKey] = migPlanLabel
```

Comment:
I think we should take care to namespace our labels that will be applied to all restored objects.

E.g. `migration.openshift.io/migrated-by-migplan` instead of just `migplan`
And `migration.openshift.io/migrated-by-migmigration` instead of `migration.openshift.io/migrated-by`